internal/services/costmanagement/export_base_resource.go

```diff
@@ -227,7 +227,13 @@ func (br costManagementExportBaseResource) updateFunc() sdk.ResourceFunc {
 				return err
 			}
 
-			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id); err != nil {
+			// Update operation requires latest eTag to be set in the request.
+			resp, err := client.Get(ctx, id.Scope, id.Name, "")
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id, resp.ETag); err != nil {
```

Comment:
The etag could be nil so could we check that beforehand? 
```suggestion
			if resp.ETag == nil {
				return fmt.Errorf("retrieving %s: etag was nil")
			}
                        
			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id, resp.ETag); err != nil {
```

Reply:
@stephybun Thank you. Fixed.